Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +849,90 @@ export class OtherSettings {
         this.resource = resource;
     }
 
-    public get editorTabSize(): number | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<number>("tabSize"); }
-    public get editorInsertSpaces(): boolean | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<boolean>("insertSpaces"); }
-    public get editorAutoClosingBrackets(): string | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<string>("autoClosingBrackets"); }
-    public get filesEncoding(): string | undefined { return vscode.workspace.getConfiguration("files", { uri: this.resource, languageId: "cpp" }).get<string>("encoding"); }
-    public get filesAssociations(): any { return vscode.workspace.getConfiguration("files").get("associations"); }
-    public set filesAssociations(value: any) {
-        void vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Workspace);
+    private getVSCodeSettingAsString(setting: string, configuration: string, resource: any, defaultString?: string): string | undefined {
+        const fullConfiguration = vscode.workspace.getConfiguration(setting, resource);
+        const value = fullConfiguration.get<string>(configuration);
```

Comment:
The other helpers name the string `settingName` and the configuration is called `setting`. Can you make these follow that pattern?

You could also consider moving those other helpers down here so that they are all close together. Just a thought.